biothings/web/query/engine.py

```diff
@@ -66,8 +66,15 @@ def __init__(self, client, indices=None):
     def execute(self, query, **options):
         assert isinstance(query, Search)
         index = self.indices[options.get('biothing_type')]
+        index = self.adjust_index()
         return self.client.search(query.to_dict(), index)
 
+    def adjust_index(self):
```

Comment:
I think you also need to pass `query, **options` to this `adjust_index` method?

Reply:
I think we could pass `query, **options` to be used.